device/common/src/make_sum_buffer.cpp

```diff
@@ -0,0 +1,48 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/device/make_sum_buffer.hpp"
+
+// System include(s).
+#include <algorithm>
+
+namespace traccc::device {
+
+TRACCC_HOST
+make_buffer_result_t<prefix_sum_size_t> make_sum_buffer(
+    const std::vector<prefix_sum_size_t> sizes,
+    const std::unique_ptr<vecmem::copy>& copy,
+    const traccc::memory_resource& mr) {
+
+    // Create vector with summation of sizes
+    vecmem::vector<device::prefix_sum_size_t> sizes_sum(
+        sizes.size(), mr.host ? mr.host : &(mr.main));
+    std::partial_sum(sizes.begin(), sizes.end(), sizes_sum.begin(),
+                     std::plus<prefix_sum_size_t>());
+    const unsigned int totalSize = sizes_sum[sizes_sum.size() - 1];
```

Comment:
I'm not a huge fan of adding this return to the current implementation. I reckon the code would be a bit cluttered as this requires functionality to be added for both cases of shared and separate memory. And if we're getting an empty input here something has already gone very wrong. Would you be unhappy if I just added an `assert(size != 0);` at the beggining?

Reply:
I'm very surprised by this comment. :confused: This particular function can and should work if it's given a jagged vector without any elements. While in our current case we don't expect that to happen, in the future we may very well have a use case for it.

And no, the implementation does not need to be complicated. Since the 1-dimensional buffer that we need to return is empty, we don't need to create anything. We can just do:

```c++
if (sizes.size() == 0) {
    return {{}, {}, 0};
}
```

The default constructors of both the vector view and the variant will be fine for this. (The default constructor of a vector view creates a view of an empty vector. Which is not pointing anywhere in memory, since an empty view does not need to point anywhere in memory.)

Reply:
I see, I was overcomplicating how this could be done. Fixed now